client/core/trade_simnet_test.go

```diff
@@ -0,0 +1,709 @@
+		return err
+	}
+
+	tLog.Infof("Trades ended at %s.", finalStatus)
+
+	// Allow some time for balance changes to be properly reported.
+	// There is usually a split-second window where a locked output
+	// has been spent but the spending tx is still in mempool. This
+	// will cause the txout to be included in the wallets locked
+	// balance, causing a higher than actual balance report.
+	time.Sleep(1 * time.Second)
+
+	for _, client := range clients {
+		if err = client.assertBalanceChanges(); err != nil {
+			return fmt.Errorf("client %d balance check error: %v", client.id, err)
+		}
+	}
+
+	return nil
+}
+
+func placeTestOrder(c *tClient, qty, rate uint64, sell bool) (string, error) {
+	dc := c.dc()
+	dcrBtcMkt := dc.market("dcr_btc")
+	if dcrBtcMkt == nil {
+		return "", fmt.Errorf("no dcr_btc market found")
+	}
+	baseAsset := dc.assets[dcrBtcMkt.BaseID]
+	quoteAsset := dc.assets[dcrBtcMkt.QuoteID]
+
+	tradeForm := &TradeForm{
+		Host:    dexHost,
+		Base:    baseAsset.ID,
+		Quote:   quoteAsset.ID,
+		IsLimit: true,
+		Sell:    sell,
+		Qty:     qty,
+		Rate:    rate,
+		TifNow:  false,
+	}
+
+	qtyStr := fmt.Sprintf("%.8f %s", fmtAmt(qty), baseAsset.Symbol)
+	rateStr := fmt.Sprintf("%.8f %s/%s", fmtAmt(rate), quoteAsset.Symbol,
+		baseAsset.Symbol)
+
+	ord, err := c.core.Trade(c.appPass, tradeForm)
+	if err != nil {
+		return "", fmt.Errorf("error placing %s order %v", sellString(sell), err)
+	}
+
+	c.log("placed order %sing %s at %s (%s)", sellString(sell), qtyStr, rateStr, ord.ID[:8])
+	return ord.ID, nil
+}
+
+func monitorTradeForTestOrder(ctx context.Context, client *tClient, orderID string, finalStatus order.MatchStatus) error {
+	errs := newErrorSet("[client %d] ", client.id)
+
+	oid, err := order.IDFromHex(orderID)
+	if err != nil {
+		return errs.add("error parsing order id %s -> %v", orderID, err)
+	}
+
+	oidShort := token(oid.Bytes())
+	client.log("waiting for matches on order %s", oidShort)
+
+	matched := tryUntil(ctx, 15*time.Second, func() bool {
+		select {
+		case n := <-client.notifications:
+			orderNote, isOrderNote := n.(*OrderNote)
+			return isOrderNote && n.Subject() == "Matches made" && orderNote.Order.ID == orderID
+		default:
+			return wait.TryAgain
+		}
+	})
+	if ctx.Err() != nil { // context canceled
+		return nil
+	}
+	if !matched {
+		return errs.add("order %s not matched after 15 seconds", oidShort)
+	}
+
+	tracker, _, _ := client.dc().findOrder(oid)
+	client.log("%d match(es) received for order %s", len(tracker.matches), oidShort)
+	for _, match := range tracker.matches {
+		client.log("%s on match %s, amount %.8f %s", match.Match.Side.String(),
+			token(match.id.Bytes()), fmtAmt(match.Match.Quantity), unbip(tracker.Base()))
+	}
+
+	// set initial match statuses before running check for status changes
+	matchStatuses := make(map[order.MatchID]order.MatchStatus, len(tracker.matches))
+	for _, match := range tracker.matches {
+		matchStatuses[match.id] = order.NewlyMatched
+	}
```

Comment:
That should probably be `lastMatchStatus` or `lastMatchStatuses`. The idea is to keep a record of the last processed match status so that if the status change checker re-sees the same status, it does not re-process it.

If the Maker is already at `MakerSwapCast`, the status change checker below will treat it as a new status report (the previous being `NewlyMatched`) and process it accordingly. The `lastMatchStatus` will also be updated to `MakerSwapCast` so the match is ignored in the next status check cycle, if the match status has not moved.